components/org.wso2.bfsi.consent.management.extensions/src/main/java/org/wso2/bfsi/consent/management/extensions/admin/utils/ConsentAdminUtils.java

```diff
@@ -0,0 +1,148 @@
+import org.wso2.bfsi.consent.management.extensions.common.ConsentException;
+import org.wso2.bfsi.consent.management.extensions.common.ConsentExtensionConstants;
+import org.wso2.bfsi.consent.management.extensions.common.ResponseStatus;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * Consent admin utils.
+ */
+public class ConsentAdminUtils {
+
+    /**
+     * Validate and retrieve query param.
+     * 1. Check whether the key exists as a query param.
+     * 2. Validate whether the value is a string.
+     * 3. Retrieve query param.
+     *
+     * @param queryParams  query params
+     * @param key          key to be retrieved
+     * @return   query param value
+     */
+    public static String validateAndGetQueryParam(Map queryParams, String key) {
+        if (queryParams.containsKey(key) && (((ArrayList<?>) queryParams.get(key)).get(0) instanceof String)) {
+            return (String) ((ArrayList<?>) queryParams.get(key)).get(0);
+        }
+        return null;
+    }
+
+    /**
+     * Get array list from query param.
+     * @param queryParam    query param values
+     * @return  array list constructed from the query param value
+     */
+    public static ArrayList<String> getArrayListFromQueryParam(String queryParam) {
+        return queryParam != null ? new ArrayList<>(Arrays.asList(queryParam.split(","))) : null;
+    }
+
+    /**
+     * Get long values from query param.
+     * @param queryParam    query param values
+     * @return  long value constructed from the query param value
+     */
+    public static long getLongFromQueryParam(String queryParam) {
+        return queryParam != null ? Long.parseLong(queryParam) : 0;
+    }
+
+    /**
+     * Get int values from query param.
+     * @param queryParam    query param values
+     * @return  int value constructed from the query param value
+     */
+    public static int getIntFromQueryParam(String queryParam) {
+        return queryParam != null ? Integer.parseInt(queryParam) : 0;
+    }
+
+    /**
+     * Convert detailed consent resource to JSON.
+     * @param detailedConsentResource   detailed consent resource
+     * @return  JSON object constructed from the detailed consent resource
+     */
+    public static JSONObject detailedConsentToJSON(DetailedConsentResource detailedConsentResource) {
+        JSONObject consentResource = new JSONObject();
+
+        consentResource.put(ConsentExtensionConstants.CC_CONSENT_ID,
+                detailedConsentResource.getConsentID());
+        consentResource.put(ConsentExtensionConstants.CLIENT_ID, detailedConsentResource.getClientID());
+        try {
+            consentResource.put(ConsentExtensionConstants.RECEIPT,
+                    new JSONObject(detailedConsentResource.getReceipt()));
+        } catch (JSONException e) {
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, "Exception occurred while parsing" +
```

Comment:
Let's add the missing throwable object, let's check other places too.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/9/commits/7640a3fd126e5e00ce693961fdc6c5cd28816bb9